.gemspec

```diff
@@ -45,7 +45,10 @@ Gem::Specification.new do |s|
   # json_pure 2.0.2+ officially requires Ruby >= 2.0, but should have specified that in 2.0
   s.add_runtime_dependency(%q<json_pure>, "~> 1.8")
   # net-ssh is a runtime dependency of Puppet::Util::NetworkDevice::Transport::Ssh
-  s.add_runtime_dependency(%q<net-ssh>, "~> 2.1")
+  # Beaker 3.0.0 to 3.10.0 depends on net-ssh 3.3.0beta1
+  # Beaker 3.11.0+ depends on net-ssh 4.0+
+  # be lenient to allow module testing where Beaker and Puppet are in same Gemfile
+  s.add_runtime_dependency(%q<net-ssh>, [">= 3.0", "< 5"]) if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.0.0')
```

Comment:
The Bundler issue I mentioned above is documented at https://github.com/bundler/bundler/pull/4651 - some more info can be found at https://stackoverflow.com/questions/40348066/can-bundler-automatically-solve-ruby-version-dependencies